tls/s2n_config.c

```diff
@@ -561,3 +596,13 @@ int s2n_config_accept_max_fragment_length(struct s2n_config *config)
     return 0;
 }
 
+int s2n_config_get_cert_type(struct s2n_config *config, s2n_cert_type *cert_type)
+{
+    notnull_check(config);
+    notnull_check(config->cert_and_key_pairs);
```

Comment:
I had considered adding `NONE` types for the optional client auth change instead of just null checking. This change might make that a cleaner choice now for things like the `s2n_set_signature_and_hash_algorithm` stuff.